cli/command/cli.go

```diff
@@ -154,8 +155,29 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
 	var err error
 	cli.client, err = NewAPIClientFromFlags(opts.Common, cli.configFile)
 	if err != nil {
-		return err
+		var (
+			passwd string
+			giveup bool
+		)
+		passRetriever := passphrase.PromptRetrieverWithInOut(cli.In(), cli.Out(), nil)
```

Comment:
Would it be OK if I put all the block under that condition? It results in one extra call to IsErrEncryptedKey, but perhaps it is more readable?

```
      if tlsconfig.IsErrEncryptedKey(err) {
                var (
                        passwd string
                        giveup bool
                )
                passRetriever := passphrase.PromptRetrieverWithInOut(cli.In(), cli.Out(), nil)

                for attempts := 0; tlsconfig.IsErrEncryptedKey(err); attempts++ {
                        // some code and comments borrowed from notary/trustmanager/keystore.go
                        passwd, giveup, err = passRetriever("private", "encrypted TLS private", false, attempts)
                        // Check if the passphrase retriever got an error or if it is telling us to give up
                        if giveup || err != nil {
                                return errors.Wrap(err, "private key is encrypted, but could not get passphrase")
                        }

                        opts.Common.TLSOptions.Passphrase = passwd
                        cli.client, err = NewAPIClientFromFlags(opts.Common, cli.configFile)
                }
        }

        if err != nil {
                return err
        }

```